### ISSUE TITLE
Sync Issue with Master Key

### ISSUE THREAD

**Issue by smhearty** [laurent22/joplin#6957]
I'm having an issue with encryption. The issue revolves around a Master Key. 

*(I believe that I may have set a different password when installing Joplin quite a while ago). I believe that I set a different password (passphrase2 for this discussion) on a Mac that I no longer use (but have access to). The problem is that I can't find a way to enter that passphrase so that I can get that key to work. For security purposes, I've redacted parts of the keys or items. I've left the beginning and end for identification.* 

When I do a sync, I get this message: 
> Created remote items: 1.
> Completed: 21/10/2022 14:36 (3s)
 Last error: Error: Could not encrypt item 019e367b1cREDACTED bce22f76525: Master key is not loaded: aa7476d89REDACTEDe4eae0cb975212

I have one key disabled. 
![Screen Shot 2022-10-21 at 14 56 18](https://user-images.githubusercontent.com/5431576/197269770-4c356d0c-bd13-4ab7-89e0-43ebb22d751d.png)

When I do re-enable it, I get the following message: 
>Master password
Master password:
✓ Valid
Your password is needed to decrypt some of your data.
Please click on "Manage master password..." to proceed, or set the passwords in the "Encryption keys" list below.

I then click on "Manage master password..." which again shows my master password to be Valid. I've clicked on Change master password and "changed" it to the same password. This doesn't work. If I change the master password to a different password, I get an error message: 
>Key aa7476d89REDACTEDe4eae0cb975212 could not be reencrypted - this is most likely due to an incorrect password. Please try again. Error was: ccm: tag doesn't match
*(*Note that I believe that this Joplin install was set up with a different password)*

Note that I use two desktops macs, iPhone and iPad in my total workflow. 

## Environment

Joplin version: 2.8.8 (Revision c2a6a13)
Platform: Mac OS
OS specifics: 12.6 (21G115)

Also using latest version of iOS. However, the issue is on one of the desktops. 

## Steps to reproduce

1. Install Joplin on a new computer. Use a different encryption passphrase
2. Sync with existing Joplin install through Dropbox
3. Pull hair out (Ok, I have a sense of humor). 

## Describe what you expected to happen
I need to resolve the sync issue. After reading through support issues, I had expected to be able to disable the key and then re-enable it with a single passphrase. 

## Logfile
>Logger.ts:219 15:07:43: Synchronizer: Error: Master key is not loaded: aa7476d89REDACTEDe4eae0cb975212
    at EncryptionService.loadedMasterKey (EncryptionService.ts:171)
    at EncryptionService.<anonymous> (EncryptionService.ts:421)
    at Generator.next (<anonymous>)
    at /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/services/e2ee/EncryptionService.js:8
    at new Promise (<anonymous>)
    at __awaiter (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/services/e2ee/EncryptionService.js:4)
    at EncryptionService.encryptAbstract_ (EncryptionService.ts:414)
    at EncryptionService.<anonymous> (EncryptionService.ts:530)
    at Generator.next (<anonymous>)
    at /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/services/e2ee/EncryptionService.js:8

I know what the second passphrase is. I just can't see a way to enter it so that I can make all of the keys the same. I'd just delete the data and application from one device if I can figure out which device that is.

**Comment by rongcuid** [comment]
Similar issue here

**Comment by smhearty** [comment]
Additional update: 

I added an iPad to the sync. On this one, I entered the Master Password for most of the targets, and then the Second Master Password for the one that has a different password. That seems to have worked. 

If I could figure out a way to reset the Master Password so that it is the same on all devices, this would probably be solved. However, I can't figure out how to do that. 

Thanks.

**Comment by smhearty** [comment]
Additional information. Sync is now pretty much completely broken. Any device that creates a new note does not sync to any of the other devices. Any changes on current notes also do not update. The error about the Master key not being loaded continues. 

I'm guessing that the way to fix this is to make a back up from one device. Then delete Joplin on each individual device. Then re-import the joplin back up back into one device. Then sync. Then wait. Then add a device. Then sync. Then wait. 

Does this seem like a path to resolution?

**Comment by smhearty** [comment]
I've tried to delete Joplin entirely from a Mac. I first turned off Syncing. I made a backup of the data. I tossed Joplin into App Cleaner. It deleted several files and folders. I then downloaded Joplin again. I opened Joplin and SURPRISE, all my data was still there. This is not what I expected. The encryption is still there as well. 
<img width="439" alt="image" src="https://user-images.githubusercontent.com/5431576/203614363-b27df946-f0e5-4a4e-9899-f929c85bf2e5.png">


I thought that I could start over and reimport my data. I must've missed something.
